### PR TITLE
feature(cb2-9608): Added stubbed endpoints for IVA defects

### PR DESCRIPTION
**Pull request by kcarr-bjss**
## Stub endpoints for IVA defects
Added stubbed endpoints for the IVA defect endpoints.

[https://dvsa.atlassian.net/browse/CB2-9608](CB2-9608)

## Checklist

- [ ] Code has been tested manually
- [ ] PR title includes the JIRA ticket number
- [ ] Branch is rebased against the latest develop
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Comment by Daniel-Searle** [comment]
Is this meant to be just a basic implementation of the "endpoints"?

**Comment by kcarr-bjss** [comment]
> Is this meant to be just a basic implementation of the "endpoints"? e.g. no type guards or validators

Yes, this is purely to establish basic endpoints so that we have something to test contracts against

File: src/config/config.yml [hunk 1]
```diff
@@ -15,6 +27,8 @@ invoke:
       getDefects:
         name: cvs-svc-defects
         mock: tests/resources/defects.json
+      getIvaDefectsByManual:
+        name: cvs-svc-defects
   remote:
```
Comment (cb-cs) [inline src/config/config.yml:32]: I know there are plans to eventually move towards SAM but whilst we are still using serverless, particularly for local invoke I think it's worth including getIvaDefects in here.

File: src/services/ivaDefectsService.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import {
+  DefectGETIVA,
+  EUVehicleCategory,
+  InspectionType,
+  VehicleType,
+} from "@dvsa/cvs-type-definitions/types/iva/defects/get/index";
+
+export class IvaDefectsService {
```
Comment (cb-cs) [inline src/services/ivaDefectsService.ts:8]: Can you refactor this filename to IvaDefectService so it follows the same convention as DefectsService.

File: src/utils/httpHeaders.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { APIGatewayProxyResult } from "aws-lambda";
```
Comment (cb-cs) [inline src/utils/httpHeaders.ts:1]: Not sure on the naming convention for this one either, should addHttpHeaders being a method in HttpHeaders class maybe? What are your thoughts?

File: src/validators/ivaDefectsByManualValidator.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
```
Comment (cb-cs) [inline src/validators/ivaDefectsByManualValidator.ts:1]: Same for this one, I wonder if it's worth having an IvaDefectsValidator class, which includes your validateIvaDefectManualQuery method, as there will likely be more validation to come.

**Comment by cb-cs** [comment]
Just a more general question, what is the test coverage like on this service following the additions? I'm just curious whether it will be over 80% in order to get through the build process on the Jenkins pipelines?

File: src/config/config.yml (hunk 1 above)
Comment (Daniel-Searle) [reply src/config/config.yml:32]: Are we ever going to want to invoke it and not just call the API?

Comment (cb-cs) [reply src/config/config.yml:34]: There may come a time when a BE service like cert-gen may want to invoke it potentially, I know for the majority of flows BE services to lambda-to-lambda invocations over API calls.

**Comment by Daniel-Searle** [comment]
> Just a more general question, what is the test coverage like on this service following the additions? I'm just curious whether it will be over 80% in order to get through the build process on the Jenkins pipelines?

Yeah, as this is an existing service it'll likely flag the coverage issue

**Pull request merged**